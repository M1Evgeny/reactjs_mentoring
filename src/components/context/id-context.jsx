import React from 'react';

const IdContext = React.createContext();

export function IdProvider(props) {
  const [movieId, setMovieId] = React.useState(0);
  const value = React.useMemo(() => ({ movieId, setMovieId }), [movieId]);
  return <IdContext.Provider value={value} {...props} />
}

export function useId() {
  const context = React.useContext(IdContext);
  if (!context) {
    throw new Error(`useId must be used within a IdProvider`);
  }
  return context;
}
