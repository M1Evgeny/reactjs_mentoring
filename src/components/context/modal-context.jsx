import React from 'react';

const ModalContext = React.createContext();

export function ModalProvider(props) {
    const [modalObject, setModalObject] = React.useState({});
    const value = React.useMemo(() => ({ modalObject, setModalObject }), [modalObject]);
    return <ModalContext.Provider value={value} {...props} />;
}

export function useModal() {
    const context = React.useContext(ModalContext);
    if (!context) {
      throw new Error(`useModal must be used within a ModalContext`);
    }
    return context; 
}
