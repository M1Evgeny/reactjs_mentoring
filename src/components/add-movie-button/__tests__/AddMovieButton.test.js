import React from "react";
import { screen, act, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AddMovieButton } from "..";
import { ModalProvider } from "../../context/modal-context";


const mockOnClick = jest.fn();

it("Current render", async () => {
  await act(async () => {
    render(
      <ModalProvider>
        <AddMovieButton onClick={mockOnClick} />
      </ModalProvider>
    );
  });
  expect(document.body).toMatchSnapshot();
});

it("Button click", async () => {
  await act(async () => {
    render(
      <ModalProvider>
        <AddMovieButton onClick={mockOnClick} />
      </ModalProvider>
    );
  });
  const addButton = screen.getByText("+ add movie");
  expect(addButton).toBeInTheDocument();

  userEvent.click(addButton);
});
