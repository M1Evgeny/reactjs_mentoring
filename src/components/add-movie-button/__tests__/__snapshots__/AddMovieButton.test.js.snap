// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current render 1`] = `
<body>
  <div>
    <div>
      <button
        class="addMovie"
        type="button"
      >
        + add movie
      </button>
    </div>
  </div>
</body>
`;
